x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/EnrichQuerySourceOperator.java

```diff
@@ -154,9 +157,11 @@ Page buildPage(int positions, IntVector.Builder positionsBuilder, IntVector.Buil
     private Query nextQuery() {
         ++queryPosition;
         while (isFinished() == false) {
-            Query query = queryList.getQuery(queryPosition);
-            if (query != null) {
-                return query;
+            if (skipMultiValuesMatching == false || queryList.block.getValueCount(queryPosition) == 1) {
```

Comment:
This `queryList.block.getValueCount(queryPosition)` is a bit weird for 2 reasons:
1. It's a protected field, accessed here for convenience
2. It's an abstract method (I'll run benchmarks to ensure this doesn't affect performance)

Reply:
I wonder if we should do this by changing `TermQueryList` - either by making a new one or pushing a boolean into it.